### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by nathanielhourt**
On my Archlinux systems, linking fails unless this change is made.

### CODE REVIEW

**Comment by ryanRfox** [comment]
Really?!?! The exact same `BOOST_COMPONENTS` in a different order?

**Comment by nathanielhourt** [comment]
Yes, but the order you pass the `-lxyz` flags to the linker does matter. Ref, for example, https://stackoverflow.com/questions/45135/why-does-the-order-in-which-libraries-are-linked-sometimes-cause-errors-in-gcc

**Review by pmconrad — APPROVED** [review]
Works for me. Thanks!

**Comment by nathanielhourt** [comment]
ICYMI, I am not able to merge this; if you guys want it, you've gotta do it yourselves. :)

**Comment by ryanRfox** [comment]
@jmjatlanta are you available to provide a second review, then merge? I trust Peter’s review, but I’d rather not be the one to merge.

**Comment by jmjatlanta** [comment]
I believe this should be sent through the develop branch first, instead of merging directly to master. I will test it there, and do another PR (Unless @nathanhourt rebases this one before I get a chance to make it)

**Review by jmjatlanta — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**
